tools/database/src/aggressive_trimming.rs

```diff
@@ -0,0 +1,194 @@
+                                r.length as usize,
+                                encode_flat_state_db_key(shard_uid, &r.hash.0),
+                            ));
+                            if keys.len() % 10000 == 0 {
+                                println!("Found {} keys to read", keys.len());
+                            }
+                        }
+                        FlatStateValue::Inlined(_) => {}
+                    }
+                },
+                true,
+            );
+        }
+
+        {
+            let non_inlined_keys = non_inlined_keys.lock().unwrap();
+            println!(
+                "Found {} non-inlined keys from FlatState, total size {}",
+                non_inlined_keys.len(),
+                non_inlined_keys.iter().map(|(size, _)| size).sum::<usize>()
+            );
+        }
+
+        {
+            let non_inlined_keys = non_inlined_keys.clone();
+            StoreParallelIterator::for_each_in_range(
+                store.clone(),
+                DBCol::FlatStateChanges,
+                Vec::new(),
+                Vec::new(),
+                6,
+                move |key: &[u8], value| {
+                    let key = KeyForFlatStateDelta::try_from_slice(key).unwrap();
+                    if key.shard_uid.version != sharding_version {
+                        return;
+                    }
+                    let changes = FlatStateChanges::try_from_slice(value).unwrap();
+                    for (_, value) in changes.0 {
+                        if let Some(FlatStateValue::Ref(r)) = value {
+                            let mut keys = non_inlined_keys.lock().unwrap();
+                            keys.push((
+                                r.length as usize,
+                                encode_flat_state_db_key(key.shard_uid, &r.hash.0),
+                            ));
+                            if keys.len() % 10000 == 0 {
+                                println!("Found {} keys to read", keys.len());
+                            }
+                        }
+                    }
+                },
+                true,
+            );
+        }
+
+        {
+            let non_inlined_keys = non_inlined_keys.lock().unwrap();
+            println!(
+                "Found {} non-inlined keys from FlatState and FlatStateChanges, total size {}",
+                non_inlined_keys.len(),
+                non_inlined_keys.iter().map(|(size, _)| size).sum::<usize>()
+            );
+        }
+
+        // Now read the non-inlined keys from the State column.
+        let non_inlined_entries = Arc::new(Mutex::new(BTreeMap::<Vec<u8>, Vec<u8>>::new()));
+        {
+            let keys =
+                non_inlined_keys.lock().unwrap().drain(..).map(|(_, key)| key).collect::<Vec<_>>();
+            let values_read = non_inlined_entries.clone();
+            StoreParallelIterator::lookup_keys(
+                store.clone(),
+                DBCol::State,
+                keys,
```

Comment:
if there are no inlined keys we will hit the assert in `WorkItem::new_lookup()`. this can actually happen and i hit this on localnet